File: Array/firstNotRepeatingCharacter.js
// returns firstNotRepeatingCharacter in a string

function firstNotRepeatingCharacter(s) {
    let obj = {};
    for(let i=0; i<s.length; i++){
        if(s[i] in obj)
            obj[s[i]]++;
        else
            obj[s[i]] = 0;
    }
    for (let key in obj) {
        if(obj[key] == 0)
            return key;
    }
    return '_';
}


console.log(firstNotRepeatingCharacter('abacabad'));
// returns c

File: Array/firstNotRepeatingCharacter.py
# returns firstNotRepeatingCharacter in a string

def firstNotRepeatingCharacter(s):
    dict = {}
    for c in s:
        if c in dict:
            dict[c] = dict[c]+1
        else:
            dict[c] = 0
    for key in dict:
        if dict[key] == 0:
            return key
    return dict


print firstNotRepeatingCharacter('abacabad')
# returns c